dartagnan/src/main/java/com/dat3m/dartagnan/encoding/PropertyEncoder.java

```diff
@@ -200,7 +201,8 @@ class SpinLoop {
         BooleanFormula atLeastOneStuck = bmgr.makeFalse();
         for (Thread t : program.getThreads()) {
             BooleanFormula isStuck = isStuckMap.getOrDefault(t, bmgr.makeFalse());
-            BooleanFormula isDone = t.getCache().getEvents(FilterBasic.get(Tag.BOUND)).stream()
+            BooleanFormula isDone = t.getCache().getEvents(FilterUnion
+            		.get(FilterBasic.get(Tag.BOUND), FilterBasic.get(Tag.EARLYTERMINATION))).stream()
```

Comment:
I think you can remove the check for `Tag.BOUND` cause you added the new tag to all bound events.